### PR TITLE
TemporalFrame changes

### PR DESCRIPTION
**Pull request by nden**

### CODE REVIEW

**Comment by nden** [comment]
@Cadair How does this look now? Please review when you get a chance. 
I removed the `reference_position` attribute as it looks like `Time.location` serves the same purpose.

File: gwcs/coordinate_frames.py [hunk 1]
```diff
@@ -382,51 +382,53 @@ class TemporalFrame(CoordinateFrame):
...
     name : str
         Name for this frame.
     """
 
-    def __init__(self, axes_order=(0,), reference_time=None,
-                 reference_frame=astropy.time.Time, unit=None,
+    def __init__(self, reference_frame, unit=None, axes_order=(0,),
                  axes_names=None, name=None, axis_physical_types=None):
 
+        axes_names = "{}({}; {}".format(reference_frame.format, reference_frame.scale, reference_frame.location)
```
Comment (Cadair) [inline gwcs/coordinate_frames.py]:
```suggestion
        axes_names = axes_names or "{}({}; {}".format(reference_frame.format, reference_frame.scale, reference_frame.location)
```

File: gwcs/coordinate_frames.py [hunk 2]
```diff
@@ -382,51 +382,53 @@ class TemporalFrame(CoordinateFrame):
...
-        `~astropy.time.Time` instance.
-    reference_time : `astropy.time.Time` or `None`
-        Reference time, the time of the 0th coordinate. If none the values of
-        the axis are assumed to be valid times.
-    unit : str or units.Unit instance
-        Spectral unit.
+    reference_frame : `~astropy.time.Time`
+        A Time object which holds the time scale and format.
+        If data is provided, it is the time zero point.
+        For relative times `data` should be an empty list.
```
Comment (Cadair) [inline gwcs/coordinate_frames.py]: I initially read this as the model returning relative times rather than the frame. Maybe reword it so it's something like "to not set a zero point for the frame"?

File: gwcs/coordinate_frames.py [hunk 3]
```diff
@@ -382,51 +382,53 @@ class TemporalFrame(CoordinateFrame):
...
 
+        axes_names = "{}({}; {}".format(reference_frame.format, reference_frame.scale, reference_frame.location)
+
         super().__init__(naxes=1, axes_type="TIME", axes_order=axes_order,
                          axes_names=axes_names, reference_frame=reference_frame,
-                         unit=unit, name=name,
-                         reference_position=reference_time, axis_physical_types=axis_physical_types)
+                         unit=unit, name=name, axis_physical_types=axis_physical_types)
+        self._attrs = {}
+        for a in self.reference_frame.info._represent_as_dict_extra_attrs:
```
Comment (Cadair) [inline gwcs/coordinate_frames.py]: Does this dict not also have the values of the attrs as well? :man_shrugging:

Comment (nden) [reply gwcs/coordinate_frames.py]: It's actually a tuple
```
t.info._represent_as_dict_extra_attrs                                                                        
Out[8]: 
('format',
 'scale',
 'precision',
 'in_subfmt',
 'out_subfmt',
 'location',
 '_delta_ut1_utc',
 '_delta_tdb_tt')

```

**Comment by Cadair** [comment]
This looks good other than the minor docstring comment. I am working on #230 and it would be easier if this was merged :smile:
